rust/src/linearview.rs

```diff
@@ -0,0 +1,515 @@
+        LinearViewLinesIterator::new(count, lines)
+    }
+}
+
+impl PartialEq for LinearViewCursor {
+    fn eq(&self, other: &Self) -> bool {
+        unsafe { BNCompareLinearViewCursors(self.handle, other.handle) == 0 }
+    }
+}
+
+impl PartialOrd for LinearViewCursor {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        match unsafe { BNCompareLinearViewCursors(self.handle, other.handle) } {
+            i if i < 0 => Some(std::cmp::Ordering::Less),
+            i if i > 0 => Some(std::cmp::Ordering::Greater),
+            _ => Some(std::cmp::Ordering::Equal),
+        }
+    }
+}
+
+impl Ord for LinearViewCursor {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match unsafe { BNCompareLinearViewCursors(self.handle, other.handle) } {
+            i if i < 0 => std::cmp::Ordering::Less,
+            i if i > 0 => std::cmp::Ordering::Greater,
+            _ => std::cmp::Ordering::Equal,
+        }
+    }
+}
+
+unsafe impl RefCountable for LinearViewCursor {
+    unsafe fn inc_ref(handle: &Self) -> Ref<Self> {
+        Ref::new(Self {
+            handle: BNNewLinearViewCursorReference(handle.handle),
+        })
+    }
+
+    unsafe fn dec_ref(handle: &Self) {
+        BNFreeLinearViewCursor(handle.handle);
+    }
+}
+
+impl ToOwned for LinearViewCursor {
+    type Owned = Ref<Self>;
+
+    fn to_owned(&self) -> Self::Owned {
+        unsafe { RefCountable::inc_ref(self) }
+    }
+}
+
+unsafe impl Send for LinearViewCursor {}
+unsafe impl Sync for LinearViewCursor {}
+
+pub struct LinearViewLinesIterator {
+    count: usize,
+    lines: *mut BNLinearDisassemblyLine,
+    offset: usize,
+}
+
+impl LinearViewLinesIterator {
+    pub(crate) fn new(count: usize, lines: *mut BNLinearDisassemblyLine) -> Self {
+        Self {
+            count,
+            lines,
+            offset: 0,
+        }
+    }
+
+    pub(crate) fn empty() -> Self {
+        Self {
+            count: 0,
+            lines: std::ptr::null_mut(),
+            offset: 0,
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.count
+    }
+}
+
+impl Iterator for LinearViewLinesIterator {
+    type Item = LinearDisassemblyLine;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.offset >= self.count {
+            return None;
+        }
+        let line = unsafe { LinearDisassemblyLine::from_raw(self.lines.offset(self.offset as _)) };
+
+        self.offset += 1;
+        Some(line)
+    }
+}
+
+
+// FIXME: we get a segmentation fault if this is there. However,
+// I don't understand enough of the API to see where this should get freed otherwise.
+/*
+impl Drop for LinearViewLinesIterator {
+    fn drop(&mut self) {
+         unsafe {
+             BNFreeLinearDisassemblyLines(self.lines, self.count);
+        }
+    }
+}
+*/
+
+pub type LinearDisassemblyLineType = BNLinearDisassemblyLineType;
+
+pub struct LinearDisassemblyLine {
+    r#type: LinearDisassemblyLineType,
```

Comment:
Just use `t`.

Reply:
Addressed in 155886564e007683342ad263b1d939fb804a954c